### DISCUSSION TITLE
Windows vulnerability not showing, but linux is

### DISCUSSION THREAD

**Discussion by skywalkerisnull** [wazuh/wazuh#18014] [Q&A] [answered]
Hello everyone,

I am playing around with Wazuh on my home lab. I am running a single node in docker version 4.4.5 and the agents are also 4.4.5
I have a ngninx proxy infront so that the FQDN has a valid cert (issued by Let's Encrypt), and I have configured all the relevant ports to be forwarded directly to the Wazuh from the router (running opnSense). The agents have all be configured to use the FQDN, and not the internal IP.

In my logs, I have the following:
```log
Jul 23, 2023 @ 14:30:25.000 wazuh-modulesd:vulnerability-detector INFO  (5450): Analyzing agent '005' vulnerabilities.
Jul 23, 2023 @ 14:30:25.000 wazuh-modulesd:vulnerability-detector INFO  (5471): Finished vulnerability assessment for agent '005'
Jul 23, 2023 @ 14:30:25.000 wazuh-modulesd:vulnerability-detector INFO  (5450): Analyzing agent '006' vulnerabilities.
Jul 23, 2023 @ 14:30:25.000 wazuh-modulesd:vulnerability-detector INFO  (5471): Finished vulnerability assessment for agent '006'
Jul 23, 2023 @ 14:30:25.000 wazuh-modulesd:vulnerability-detector INFO  (5472): Vulnerability scan finished.
Jul 23, 2023 @ 14:30:24.000 wazuh-modulesd:vulnerability-detector INFO  (5431): Starting vulnerability scan.
Jul 23, 2023 @ 14:30:24.000 wazuh-modulesd:vulnerability-detector INFO  (5450): Analyzing agent '000' vulnerabilities.
Jul 23, 2023 @ 14:30:24.000 wazuh-modulesd:vulnerability-detector INFO  (5471): Finished vulnerability assessment for agent '000'
Jul 23, 2023 @ 14:30:24.000 wazuh-modulesd:vulnerability-detector INFO  (5450): Analyzing agent '002' vulnerabilities.
Jul 23, 2023 @ 14:30:24.000 wazuh-modulesd:vulnerability-detector INFO  (5471): Finished vulnerability assessment for agent '002'
Jul 23, 2023 @ 14:30:24.000 wazuh-modulesd:vulnerability-detector INFO  (5450): Analyzing agent '003' vulnerabilities.
Jul 23, 2023 @ 14:30:24.000 wazuh-modulesd:vulnerability-detector INFO  (5471): Finished vulnerability assessment for agent '003'
Jul 23, 2023 @ 14:30:24.000 wazuh-modulesd:vulnerability-detector INFO  (5450): Analyzing agent '004' vulnerabilities.
Jul 23, 2023 @ 14:30:24.000 wazuh-modulesd:vulnerability-detector INFO  (5471): Finished vulnerability assessment for agent '004'
```

![image](https://github.com/wazuh/wazuh/assets/7260106/4d681cde-11af-449c-a932-856f184a0a97)

Windows machine:
![image](https://github.com/wazuh/wazuh/assets/7260106/a9d38654-b450-47c2-9097-13c32f432d90)

Linux machine:
![image](https://github.com/wazuh/wazuh/assets/7260106/3718f029-e91c-432d-a203-55807f144caf)

Vulnerability has been configured:
```xml
  <vulnerability-detector>
    <enabled>yes</enabled>
    <interval>5m</interval>
    <min_full_scan_interval>6h</min_full_scan_interval>
    <run_on_start>yes</run_on_start>

    <!-- Ubuntu OS vulnerabilities -->
    <provider name="canonical">
      <enabled>yes</enabled>
      <os>bionic</os>
      <os>focal</os>
      <os>jammy</os>
      <update_interval>1h</update_interval>
    </provider>

    <!-- Debian OS vulnerabilities -->
    <provider name="debian">
      <enabled>no</enabled>
      <os>stretch</os>
      <os>buster</os>
      <os>bullseye</os>
      <update_interval>1h</update_interval>
    </provider>

    <!-- RedHat OS vulnerabilities -->
    <provider name="redhat">
      <enabled>no</enabled>
      <os>5</os>
      <os>6</os>
      <os>7</os>
      <os>8</os>
      <update_interval>1h</update_interval>
    </provider>

    <!-- Amazon Linux OS vulnerabilities -->
    <provider name="alas">
      <enabled>no</enabled>
      <os>amazon-linux</os>
      <os>amazon-linux-2</os>
      <update_interval>1h</update_interval>
    </provider>

    <!-- Arch OS vulnerabilities -->
    <provider name="arch">
      <enabled>no</enabled>
      <update_interval>1h</update_interval>
    </provider>

    <!-- Windows OS vulnerabilities -->
    <provider name="msu">
      <enabled>yes</enabled>
      <update_interval>1h</update_interval>
    </provider>

    <!-- Aggregate vulnerabilities -->
    <provider name="nvd">
      <enabled>yes</enabled>
      <update_from_year>2010</update_from_year>
      <update_interval>1h</update_interval>
    </provider>

  </vulnerability-detector>
```

Does anyone have any ideas of what I am missing and can change to make the windows vulnerabilities show?

**Comment by mdiego92** [comment]
Hello @skywalkerisnull,

Could you please confirm if you have the syscollector wodle enabled in your Windows agent? You can also override the configuration with a centralized groups configuration for that agent.

E.g.:
```
<wodle name="syscollector">
   <disabled>no</disabled>
   <interval>1h</interval>
   <os>yes</os>
   <packages>yes</packages>
   <hotfixes>yes</hotfixes>
</wodle>
```

Documentation for reference: https://documentation.wazuh.com/current/user-manual/capabilities/vulnerability-detection/configuring-scans.html

Please let me know if you were able to get vulnerabilities from your Windows endpoint.
Looking forward to your feedback!

**Reply by skywalkerisnull** [reply]
@mdiego92 In the manager config I have the following:

```xml
  <wodle name="syscollector">
    <disabled>no</disabled>
    <interval>1h</interval>
    <scan_on_start>yes</scan_on_start>
    <hardware>yes</hardware>
    <os>yes</os>
    <network>yes</network>
    <packages>yes</packages>
    <hotfixes>yes</hotfixes>
    <ports all="yes">yes</ports>
    <processes>yes</processes>

    <!-- Database synchronization settings -->
    <synchronization>
      <max_eps>10</max_eps>
    </synchronization>
  </wodle>
```
I have a custom group for the windows devices, but that only specifies a few additional locations for logs that we want picked up.

And I confirmed that the same was also in the `ossec.conf` on one of the windows machines. Just to confirm, I removed the `<hotfixes>yes</hotfixes>` restarted the server, then manually removed it from the windows machine (restarted the wazuh service). Left it for about an hour, and then re-added it. And the vulnerabilities are still not appearing in the Wazuh dashboard for the windows machine.

**Reply by mdiego92** [reply]
Hello @skywalkerisnull,

Is this issue happening in multiple Windows agents? It could be possible that the Windows agent is fully upgraded, hence not showing any vulnerabilities. 


Even though, I would recommend checking it by enabling the debugging module:

- Could you please modify the `/var/ossec/etc/local_internal_options.conf` file by adding the following line?

`wazuh_modules.debug=2`

- Then, please restart the manager and send me the output of the following command:

`cat /var/ossec/logs/ossec.log | grep -i 'vuln'
`

- Just in case, we can do the same procedure for the agent, by adding `windows.debug=2` in the `C:\Program Files (x86)\ossec-agent/local_internal_options.conf` file. After that, restart the agent service and send us the `ossec.log` file.

**Please remember to revert these changes by removing the lines and restarting both the manager and agent, as these debugging options will be generating more logs and consume more resources.** 

Looking forward to your response!
Best regards!

**Reply by skywalkerisnull** [reply]
Hey @mdiego92 

It is all the windows devices, a mixture of win10 and win11. The win 11's are updated pretty quickly, and I can believe that Windows on those would not have vulnerabilities, but the win 10 devices are slow to get patched.

I will have a look at the logs over the weekend and post back with the details.

**Reply by mdiego92** [reply]
Hello @skywalkerisnull,

Thank you for the quick feedback. I'll be waiting for your response!
Have a great weekend.

**Reply by skywalkerisnull** [reply]
@mdiego92

From the server:
https://gist.github.com/skywalkerisnull/d1db45caa25e90a35a32ce48286eba21

From one of the windows devices:
https://gist.github.com/skywalkerisnull/f1daad5201bf0114d467592a84fc1d5a

**Reply by skywalkerisnull** [reply]
Not sure if this is related, but I created a new group to collect the performance metrics from the windows machines following this: https://wazuh.com/blog/monitoring-windows-resources-with-performance-counters/ added all of the window's devices to the group, and restarted the Wazuh service (also added the wazuh_command.remote_commands=1 to the agent side)

When I inspect the ossec.conf file manually on the windows machines (after multiple reboots, and service restarts) the new commands are not in the config, and when in inspect the configuration in the Wazuh dashboard, there are no commands present:

![image](https://github.com/wazuh/wazuh/assets/7260106/1e201999-6aa8-4582-b63e-b27057e925c0)

**Reply by mdiego92** [reply]
Hello @skywalkerisnull,

Thank you for the update on this.
In Windows agents, the centralized configuration is placed in the `C:\Program Files (x86)\ossec-agent\shared\agent.conf` file.

Regarding the Windows Server 2012 environment, could you please verify if you can see your agent's hotfixes and installed packages inventory?

You can check that from Inventory Data at the top right corner:

![image](https://github.com/wazuh/wazuh/assets/62477966/2c3b51e1-8251-4200-96f8-37411bdfb072)

If you prefer to get said information without accessing the user interface, you can run these commands (remember to replace the 00X with your agent ID):

```
sqlite3 /var/ossec/queue/db/00X.db "select * from sys_programs;"
sqlite3 /var/ossec/queue/db/00X.db "select * from sys_hotfixes;"
```

-

In my case, I have all the available updates:

![image](https://github.com/wazuh/wazuh/assets/62477966/4a5d611c-9f6d-42ac-bc32-14531a488829)

And I installed an older version of Python, to make sure the vulnerabilities are detected:

![image](https://github.com/wazuh/wazuh/assets/62477966/a402112a-e59c-4053-85f9-5c0e85dc2c9a)

Looking forward to your feedback!

Best regards.

**Reply by skywalkerisnull** [reply]
Hello @mdiego92 

I have installed Python 2.7 and the vulnerabilities are showing up!

![image](https://github.com/wazuh/wazuh/assets/7260106/0de3d341-7ecc-4b18-8d0a-f8a50ff3e5e2)

Ok, so this must mean that our test machines are too clean.... Time to find some targets that are less clean

**Reply by mdiego92** [reply, marked as answer]
Hello @skywalkerisnull,

It seems it is working as expected!
Thank you very much for the feedback :smile:
